### ISSUE TITLE
Thanos ruler is not working as expected

### ISSUE THREAD

**Issue by monica0500** [thanos-io/thanos#5871]
I am trying to enable Thanos stateless ruler via remote-write following the below documents:
https://thanos.io/tip/components/rule.md/#stateless-ruler-via-remote-write

I am using a configmap in our Kubernetes environment for loading the yaml configuration for remote write, here is the sample content:

remote_write:
  - url: "http://<domain>:<port>/prometheus/api/v1/write"
    name: thanos-receiver
    follow_redirects: true
    remote_timeout: 30s

and in my thanos-ruler args I added --remote-write.config-file with metioning the path to the above yaml which is accessible.
When I restart the pods,  I get the below error:

level=error ts=2022-11-07T08:03:40.550344914Z caller=main.go:132 err="url for remote_write is empty\ngithub.com/prometheus/prometheus/config.(*RemoteWriteConfig).UnmarshalYAML\n\t/opt/go/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20211119115433-692a54649ed7/config/config.go:705\ngopkg.in/yaml%2ev2.(*decoder).callUnmarshaler\n\t/opt/go/pkg/mod/gopkg.in/yaml.v2@v2.4.0/decode.go:270\ngopkg.in/yaml%2ev2.(*decoder).prepare\n\t/opt/go/pkg/mod/gopkg.in/yaml.v2@v2.4.0/decode.go:313\ngopkg.in/yaml%2ev2.

Please suggest a solution.

**Comment by michael-burt** [comment]
You have not provided a URL for Thanos Receiver, this is not valid: `url: "http://:/prometheus/api/v1/write"`

it probably should look more like: `url: "http://my-receive-service-name.my-namespace:9090/prometheus/api/v1/write"`

**Comment by matej-g** [comment]
@monica0500 can you please properly format and correct the config you provided? As @michael-burt this will not work. I just tested the remote write config and all seems to work fine. Please make sure you format your YAML properly.

**Comment by monica0500** [comment]
Hi, The url seems to got updated when I added the comment, here is the URL," http://domainname:8080/prometheus/api/v1/write", I url which I am giving I tested through curl command and it was working fine.

**Comment by monica0500** [comment]
Here are the versions of Prometheus and thanos:
thanos:v0.24.0-b1.0
prometheus:v2.34.0-b3.0

**Comment by michael-burt** [comment]
I recommend confirming that the configuration file does include the correct url by inspecting it on the running Thanos Ruler pod. Have you done that yet?

**Comment by monica0500** [comment]
Yes, I have done that. Can you send me some result we should get after running the url?

**Comment by michael-burt** [comment]
> Can you send me some result we should get after running the url?

I am not sure what you mean by this. Can you paste your full config in a code block?

**Comment by monica0500** [comment]
Here is the below configuration:
  ` name: "thanos01"
    remote_write:
    - url: "http://domainName:8080/prometheus/api/v1/write"
      name: "thanos"
      follow_redirects: "true"
      remote_timeout: "30s"`

What did you meant by " url by inspecting it on the running Thanos Ruler pod"? Also after inspecting that what results you get, wanted to check if we are on the same page so needed some sample.

**Comment by monica0500** [comment]
Can you please reply on the above?

**Comment by matej-g** [comment]
Hey @monica0500, please take a look at the quickstart script (https://github.com/thanos-io/thanos/blob/main/scripts/quickstart.sh#L262), I have adjusted it and made sure it works. If you follow the example, you should be able to configure the ruler. We should then close this issue, since it seems ruler is behaving as expected but the issue is rather on your side.

**Comment by monica0500** [comment]
Hey @matej-g I tried the above but it's still giving issues, can you please let me know how can I validate the url? I am suspecting some issue with the url.

**Comment by monica0500** [comment]
Hi, Any updates?

**Comment by yeya24** [comment]
I don't see any issue running with the stateless mode. Please share the detailed error log and your config file and format it correctly using some yaml formatter https://jsonformatter.org/yaml-formatter.
Otherwise it is really hard for us to debug the problem you encountered.
